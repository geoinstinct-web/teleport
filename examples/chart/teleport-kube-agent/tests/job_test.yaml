suite: Job
templates:
  - delete_hook.yaml

release:
  upgrade: true
tests:
  - it: should set securityContext in post-delete hook
    template: delete_hook.yaml
    # first 3 items are ServiceAccount, Role, RoleBinding, 3rd (zero-indexed) is the Job
    documentIndex: 3
    values:
      - ../.lint/backwards-compatibility.yaml
    asserts:
      - equal:
          path: spec.template.spec.containers[0].securityContext.allowPrivilegeEscalation
          value: false
      - equal:
          path: spec.template.spec.containers[0].securityContext.capabilities
          value:
            drop:
              - all
      - equal:
          path: spec.template.spec.containers[0].securityContext.readOnlyRootFilesystem
          value: true
      - equal:
          path: spec.template.spec.containers[0].securityContext.runAsNonRoot
          value: true
      - equal:
          path: spec.template.spec.containers[0].securityContext.runAsUser
          value: 9807
      - matchSnapshot:
          path: spec.template.spec

  - it: should set nodeSelector in post-delete hook
    template: delete_hook.yaml
    # first 3 items are ServiceAccount, Role, RoleBinding, 3rd (zero-indexed) is the Job
    documentIndex: 3
    values:
      - ../.lint/node-selector.yaml
    asserts:
      - equal:
          path: spec.template.spec.nodeSelector
          value:
            gravitational.io/k8s-role: node
      - matchSnapshot:
          path: spec.template.spec
