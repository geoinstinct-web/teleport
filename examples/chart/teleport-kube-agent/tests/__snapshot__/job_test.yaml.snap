should set nodeSelector in post-delete hook:
  1: |
    containers:
    - args:
      - kube-state
      - delete
      command:
      - teleport
      env:
      - name: KUBE_NAMESPACE
        valueFrom:
          fieldRef:
            fieldPath: metadata.namespace
      - name: RELEASE_NAME
        value: RELEASE-NAME
      image: 'public.ecr.aws/gravitational/teleport:'
      imagePullPolicy: IfNotPresent
      name: post-delete-job
      securityContext:
        allowPrivilegeEscalation: false
        capabilities:
          drop:
          - all
        readOnlyRootFilesystem: true
        runAsNonRoot: true
        runAsUser: 9807
    nodeSelector:
      gravitational.io/k8s-role: node
    restartPolicy: OnFailure
    serviceAccountName: RELEASE-NAME-delete-hook
should set securityContext in post-delete hook:
  1: |
    containers:
    - args:
      - kube-state
      - delete
      command:
      - teleport
      env:
      - name: KUBE_NAMESPACE
        valueFrom:
          fieldRef:
            fieldPath: metadata.namespace
      - name: RELEASE_NAME
        value: RELEASE-NAME
      image: 'public.ecr.aws/gravitational/teleport:'
      imagePullPolicy: IfNotPresent
      name: post-delete-job
      securityContext:
        allowPrivilegeEscalation: false
        capabilities:
          drop:
          - all
        readOnlyRootFilesystem: true
        runAsNonRoot: true
        runAsUser: 9807
    restartPolicy: OnFailure
    serviceAccountName: RELEASE-NAME-delete-hook
